<launch>
  <arg name="input/pointcloud" default="/sensing/lidar/top/pointcloud_raw_ex"/>
  <arg name="input/imu" default="/sensing/imu/imu_data"/>
  <arg name="input/twist" default="/sensing/vehicle_velocity_converter/twist_with_covariance"/>
  <arg name="output/pointcloud" default="/sensing/lidar/top/test"/>

  <arg name="cuda_pointcloud_preprocessor_param_file" default="$(find-pkg-share autoware_cuda_pointcloud_preprocessor)/config/cuda_pointcloud_preprocessor.param.yaml"/>

  <node pkg="autoware_cuda_pointcloud_preprocessor" exec="cuda_pointcloud_preprocessor_node" name="cuda_pointcloud_preprocessor" output="screen">
    <remap from="~/input/pointcloud" to="$(var input/pointcloud)"/>
    <remap from="~/input/imu" to="$(var input/imu)"/>
    <remap from="~/input/twist" to="$(var input/twist)"/>
    <remap from="~/output/pointcloud" to="$(var output/pointcloud)"/>
    <param from="$(var cuda_pointcloud_preprocessor_param_file)"/>
  </node>
</launch>
